odbc/src/api/data.rs

```diff
@@ -1341,7 +1341,7 @@ mod unit {
                             $input
                         );
                         if $info.is_some() {
-                            let odbc_info = r.1.unwrap().get_sql_state().to_string();
+                            let odbc_info = r.1.unwrap().get_sql_state(OdbcVersion::Odbc3_80).to_string();
```

Comment:
I am wondering what you think about having get_sql_state return an OdbcState and only get the correct code for the odbc version in get_diag_recw and get_diag_field.
This way you shouldn't have to worry the odbc_version in unit test. Checking that the error has the correct OdbcState would be enough.

I don't think this is mandatory however, because it seems that the validation of "raw" Error in unit tests is very limited.